src/RoomNotifs.ts

```diff
@@ -81,9 +81,12 @@ export function setRoomNotifsState(client: MatrixClient, roomId: string, newStat
 }
 
 export function getUnreadNotificationCount(room: Room, type: NotificationCountType, threadId?: string): number {
-    let notificationCount = !!threadId
-        ? room.getThreadUnreadNotificationCount(threadId, type)
-        : room.getUnreadNotificationCount(type);
+    const countShownForRoom = (): number =>
+        SettingsStore.getValue("threadsActivityCentre")
```

Comment:
Settings isn't very performant, could we pass this value in and only read it periodically? I presume this method is called significantly

Reply:
Hmm... kinda. I'm considering whether it would be simpler to memoise settings getters...

Reply:
ok, done